blazor/treegrid/rows/detail-template.md

```diff
@@ -0,0 +1,177 @@
+        <TreeGridColumn Field="Country" HeaderText="Country" Width="100"></TreeGridColumn>
+    </TreeGridColumns>
+</SfTreeGrid>
+
+
+@code{
+    public Employee model = new Employee();
+    public IEnumerable<Employee> TreeData { get; set; }
+    protected override void OnInitialized()
+    {
+        this.TreeData = Employee.GetTemplateData();
+    }
+}
+
+{% endhighlight %}
+
+{% highlight c# %}
+
+namespace TreeGridComponent.Data {
+
+public class Employee
+    {
+        public string Name { get; set; }
+        public string FullName { get; set; }
+        public DateTime? DOB { get; set; }
+        public string Designation { get; set; }
+        public string EmpID { get; set; }
+        public int? EmployeeID { get; set; }
+        public string Country { get; set; }
+        public string Address { get; set; }
+        public string Contact { get; set; }
+        public int? ParentId { get; set; }
+        public TreeData Treedata { get; set; }
+
+        public static List<Employee> GetTemplateData()
+        {
+            List<Employee> DataCollection = new List<Employee>();
+
+            DataCollection.Add(new Employee { Name = "Robert King",FullName = "RobertKing",Designation = "Chief Executive Officer",EmployeeID = 1,EmpID = "EMP001",Address = "507 - 20th Ave. E.Apt. 2A, Seattle",Contact = "(206) 555-9857",Country = "USA",DOB = new DateTime(1963, 2, 15),ParentId = null,Treedata = new TreeData() { ID = 21}});
+            DataCollection.Add(new Employee { Name = "David william",FullName = "DavidWilliam",Designation = "Vice President",EmployeeID = 2,EmpID = "EMP004",Address = "722 Moss Bay Blvd., Kirkland",Contact = "(206) 555-3412",Country = "USA",DOB = new DateTime(1971, 5, 20),ParentId = 1,Treedata = new TreeData() { ID = 21 }});
+            DataCollection.Add(new Employee { Name = "Nancy Davolio",FullName = "NancyDavolio",Designation = "Marketing Executive",EmployeeID = 3,EmpID = "EMP035",Address = "4110 Old Redmond Rd., Redmond",Contact = "(206) 555-8122",Country = "USA",DOB = new DateTime(1966, 3, 19),ParentId = 1,Treedata = new TreeData() { ID = 21 }});
+            DataCollection.Add(new Employee { Name = "Andrew Fuller",FullName = "AndrewFuller",Designation = "Sales Representative",EmployeeID = 4,EmpID = "EMP045",Country = "UK",DOB = new DateTime(1980, 9, 20),ParentId = 1,Treedata = new TreeData() { ID = 21 }});
+            return DataCollection;
+        }
+    }
+}
+
+{% endhighlight %}
+
+{% endtabs %}
+
+![Blazor Tree Grid with Detail Template](../images/blazor-treegrid-detail-template.png)
+
+## Rendering custom component
+
+To render the custom component inside the detail row, define the template in the [DetailTemplate](../templates/#detailtemplates-component) and render the custom component in any of the detail row element.
+
+In the following sample, tree grid component is rendered as a detailed template of the row.

```

Comment:
The tree grid component is rendered as a detailed row template in the following sample.